### PR TITLE
Added material-ui, replaced react-mdl Dialog with material-ui Dialog

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Required for #277

#### What's this PR do?
There's an issue with the material-ui AutoComplete because it uses z-index to hover above the textfield area. The HTML5 dialog element which react-mdl uses is defined as top-layer when activated using `dialog.showModal()` so we can't use it with the AutoComplete. Fortunately the material-ui Dialog is a drop in replacement

#### How should this be manually tested?
Make sure the dialogs still work. There's one in App.js and one in the education tab when you click the edit or plus button. Cancel or ESC or clicking outside the dialog should cancel, and the other buttons should act like before.

#### Any background context you want to provide?

The react-select component will be truncated within the textbox, but we're getting rid of it in #277 anyway

### CODE REVIEW

**Comment by aliceriot** [comment]
there are a bunch of css changes that appear to be just code cleanup, is that correct?

**Comment by noisecapella** [comment]
Not exactly, it removes the surrounding `#dashboard { }`. material-ui puts the dialog at the body level so the CSS needs to be modified to handle those fields correctly.

**Comment by aliceriot** [comment]
ah gotcha, that makes sense.

File: static/js/containers/App.js [hunk 1]
```diff
@@ -23,7 +18,6 @@ import {
 } from '../actions/index';
 import {
   validateProfileComplete,
-  doDialogPolyfill
 } from '../util/util';
```
Comment (aliceriot) [inline static/js/containers/App.js]: nit: this could probably go on one line now

**Comment by aliceriot** [comment]
aside from that one comment this looks good to me! :+1:

**Pull request merged**